### PR TITLE
Production/GFS.v16: update wcoss2 module to use versions

### PR DESCRIPTION
**Pull request by junwang-noaa**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model.

## Description

This PR is to update the production/GFS.v16 to use the build_version and run_version in wcoss2 module file

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!)
- https://github.com/ufs-community/ufs-weather-model/issues/1424

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [ ] hera.gnu
- [x] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [x] wcoss2.intel
- [ ] acorn.intel
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

- Depends on: https://github.com/NOAA-EMC/NEMS/pull/114

### CODE REVIEW

File: modulefiles/wcoss2/fv3 [hunk 1]
```diff
@@ -1,53 +1,95 @@
...
-setenv CMAKE_CXX_COMPILER CC
-setenv CMAKE_Fortran_COMPILER ftn
-setenv CMAKE_Platform wcoss2
+help([[
+loads NEMS FV3 prerequisites on Cactus and Dogwood
+]])
+
+PrgEnv_intel_ver=os.getenv("PrgEnv_intel_ver") or "8.1.0"
+load(pathJoin("PrgEnv-intel", PrgEnv_intel_ver))
+--module load PrgEnv-intel/8.1.0
```
Comment (DusanJovic-NOAA) [inline modulefiles/wcoss2/fv3]: I would remove these commented 'module load ..' commands.

**Review by DusanJovic-NOAA — APPROVED** [review]

File: modulefiles/wcoss2/fv3 (hunk 1 above)
Comment (junwang-noaa) [reply modulefiles/wcoss2/fv3]: all the module version comments are removed

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by aerorahul — COMMENTED** [review]
I tested these on WCOSS2.
I am not sure how this will work on Hera/Orion, since the modulefiles for those machines are deleted in this PR?

**Review by aerorahul — APPROVED** [review]
yes.  This looks right now for Hera and Orion.

**Pull request merged**